src/renderer/components/dock/dock-tab.store.ts

```diff
@@ -34,14 +49,18 @@ export class DockTabStore<T = any> {
     });
   }
 
-  protected serializeData() {
-    const { storageSerializer } = this.options;
+  protected serializeBeforeSave(data: T): T {
+    return data;
+  }
 
-    return Array.from(this.data).map(([tabId, tabData]) => {
-      if (storageSerializer) return [tabId, storageSerializer(tabData)];
+  protected getStorableData(): DockTabStorageState<T> {
+    const allTabsData = this.data.toJSON();
 
-      return [tabId, tabData];
-    });
+    return Object.entries(allTabsData).reduce((data, [tabId, tabData]) => {
+      data[tabId] = this.serializeBeforeSave(tabData);
+
+      return data;
+    }, allTabsData);
```

Comment:
I would recommend recreating it.
```suggestion
    return Object.fromEntries(
      Object.entries(allTabsData)
        .map(([tabId, tabData]) => ([
          tabId,
          this.finalizeDataForSave(tabData),
        ])
    );
```